cmd/metal-api/internal/grpc/grpcServer.go

```diff
@@ -0,0 +1,191 @@
+package grpc
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	grpc_zap "github.com/grpc-ecosystem/go-grpc-middleware/logging/zap"
+	grpc_recovery "github.com/grpc-ecosystem/go-grpc-middleware/recovery"
+	grpc_ctxtags "github.com/grpc-ecosystem/go-grpc-middleware/tags"
+	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
+	"github.com/metal-stack/masterdata-api/pkg/interceptors/grpc_internalerror"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/keepalive"
+	"io/ioutil"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/metal-stack/metal-api/cmd/metal-api/internal/metal"
+	v1 "github.com/metal-stack/metal-api/pkg/api/v1"
+	"github.com/metal-stack/metal-lib/bus"
+	"github.com/metal-stack/metal-lib/zapup"
+	"go.uber.org/zap"
+)
+
+const (
+	receiverHandlerTimeout  = 15 * time.Second
+	allocationTopicTTL      = time.Duration(30) * time.Second
+	defaultResponseInterval = 5 * time.Second
+	defaultCheckInterval    = 60 * time.Second
+)
+
+func timeoutHandler(err bus.TimeoutError) error {
+	zapup.MustRootLogger().Sugar().Error("Timeout processing event", "event", err.Event())
+	return nil
+}
+
+type Datasource interface {
+	FindMachineByID(machineID string) (*metal.Machine, error)
+	UpdateMachine(old, new *metal.Machine) error
+}
+
+type ServerConfig struct {
+	Publisher             bus.Publisher
+	Datasource            Datasource
+	Logger                *zap.SugaredLogger
+	NsqTlsConfig          *bus.TLSConfig
+	NsqlookupdHttpAddress string
```

Comment:
Just from the wording it does not sound odd to me that these things are part of a "server config". 

I would appreciate if we could move service implementations into separate files and these services have their own constructors. At least to me this would make the code much easier to understand from the outside.

Service construction would need to be done from `main.go` in the main init process.

Reply:
Good point, readability should always be improved as much as possible